checkov/common/checks_infra/solvers/attribute_solvers/base_attribute_solver.py

```diff
@@ -19,11 +21,17 @@ def __init__(self, resource_types: List[str], attribute: Optional[str], value: A
         self.value = value
 
     def run(self, graph_connector: DiGraph) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]:
-        all_vertices_resource_types = [
-            data for _, data in graph_connector.nodes(data=True) if self.resource_type_pred(data, self.resource_types)
-        ]
-        passed_vertices = [data for data in all_vertices_resource_types if self.get_operation(vertex=data)]
-        failed_vertices = [resource for resource in all_vertices_resource_types if resource not in passed_vertices]
+        jobs = []
+        passed_vertices = []
+        failed_vertices = []
+        for _, data in graph_connector.nodes(data=True):
+            thread = threading.Thread(target=self._process_node(data, passed_vertices, failed_vertices))
+            jobs.append(thread)
+            thread.start()
+
+        for job in jobs:
+            job.join()
```

Comment:
I would use https://docs.python.org/3/library/concurrent.futures.html instead of the more raw `threading` API. From the docs it is more thread safe and efficient on memory management, and with less code